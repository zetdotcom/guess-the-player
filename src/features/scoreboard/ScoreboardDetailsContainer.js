import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { WINNING_SCORE } from '../../constants';
import { setWinner } from '../scoreboard/scoreboardSlice';

function ScoreboardDetailsContainer() {
	const dispatch = useDispatch();
	const { score, attempts } = useSelector((state) => state.scoreboard);

	useEffect(
		() => {
			score >= WINNING_SCORE && dispatch(setWinner());
		},
		// eslint-disable-next-line react-hooks/exhaustive-deps
		[ score ]
	);

	return (
		<div>
			<div>
				Score: {score} / {WINNING_SCORE}
			</div>
			<div>Attempts: {attempts}</div>
			<div>Accuracy: {attempts > 0 && (score / attempts * 100).toFixed() + '%'}</div>
		</div>
	);
}

export default ScoreboardDetailsContainer;
