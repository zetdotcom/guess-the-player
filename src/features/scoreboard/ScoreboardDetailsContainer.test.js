import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import store from '../../redux/store';
import ScoreboardDetailsContainer from './ScoreboardDetailsContainer';

afterEach(cleanup);

const renderComponent = newStore =>
	render(
		<Provider store={store(newStore)}>
			<ScoreboardDetailsContainer />
		</Provider>
	);

describe('TEST ScoreboardDetailsContainer', () => {
	test('should match snaptshot', () => {
		const { asFragment } = renderComponent();

		expect(asFragment).toMatchSnapshot();
	});

	test('should be all zeros in initial scoreboard', () => {
		const { getByTestId } = renderComponent();

		const score = getByTestId('score');
		const attempts = getByTestId('attempts');
		const accuracy = getByTestId('accuracy');
		expect(score).toHaveTextContent('Score: 0 / 10');
		expect(attempts).toHaveTextContent('Attempts: 0');
		expect(accuracy).toHaveTextContent('Accuracy:');
	});

	test('should show correct scoreboard results', () => {
		const newState = {
			scoreboard: {
				score: 5,
				attempts: 10,
			},
		};
		const { getByTestId } = renderComponent(newState);

		const score = getByTestId('score');
		const attempts = getByTestId('attempts');
		const accuracy = getByTestId('accuracy');
		expect(score).toHaveTextContent('Score: 5 / 10');
		expect(attempts).toHaveTextContent('Attempts: 10');
		expect(accuracy).toHaveTextContent('Accuracy: 50%');
	});
});
