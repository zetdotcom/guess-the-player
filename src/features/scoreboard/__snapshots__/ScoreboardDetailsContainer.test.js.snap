// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST ScoreboardDetailsContainer should match snaptshot 1`] = `[Function]`;
